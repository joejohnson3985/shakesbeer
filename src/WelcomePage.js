import React, { Component } from 'react';
import Autocomplete from './Autocomplete.js';
import './WelcomePage.css';

class WelcomePage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      usStates: this.props.dataset.reduce((acc, brewery) => {
        if(!acc.includes(brewery.state)) {
          acc.push(brewery.state)
        }
        return acc;
      }, [])
    }
  }

  // getUsStates() {
  //   const states = this.props.dataset.reduce((acc, brewery) => {
  //     if(!acc.includes(brewery.state)) {
  //       acc.push(brewery.state)
  //     }
  //     return acc;
  //   }, [])
  //   this.setState({ usStates: states});
  // }
  

  render() {
    return (
      <div className='welcome-page'>
        <h1 className='welcome-header'>ShakesBeer</h1>
        <div className='search-container'>
          <Autocomplete suggestions={this.state.usStates} 
                        dataset={this.props.dataset} 
                        chooseState={this.props.chooseState}
          />
        </div>
      </div>
    );
  }
}

export default WelcomePage;