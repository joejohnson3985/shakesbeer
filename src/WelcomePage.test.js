import React from 'react';
import WelcomePage from './WelcomePage.js';
import { shallow } from 'enzyme';

const mockData = [1,2,3,4];
const mockFunc = jest.fn();

describe('WelcomePage', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <WelcomePage dataset={mockData} 
                          chooseState={mockFunc}
             />
        )
    });

    it('Should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });
});