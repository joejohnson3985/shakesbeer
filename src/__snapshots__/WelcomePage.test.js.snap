// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage Should match the snapshot 1`] = `
<div
  className="welcome-page"
>
  <h1
    className="welcome-header"
  >
    ShakesBeer
  </h1>
  <div
    className="search-container"
  >
    <Autocomplete
      chooseState={[MockFunction]}
      dataset={
        Array [
          1,
          2,
          3,
          4,
        ]
      }
      suggestions={
        Array [
          undefined,
        ]
      }
    />
  </div>
</div>
`;
